paddle/fluid/operators/concat_op.cc

```diff
@@ -153,6 +156,33 @@ class ConcatGradOpMaker : public framework::SingleGradOpMaker<T> {
   }
 };
 
+class ConcatCompositeGradOpMaker : public prim::CompositeGradOpMakerBase {
+  using prim::CompositeGradOpMakerBase::CompositeGradOpMakerBase;
+
+ public:
+  void Apply() override {
+    std::vector<paddle::experimental::Tensor> input =
+        this->GetMultiForwardInput("X");
+    paddle::experimental::Tensor out_grad = this->GetSingleOutputGrad("Out");
+    std::vector<paddle::experimental::Tensor> input_grad =
+        this->GetMultiForwardInput("X");
+
+    std::vector<paddle::experimental::Tensor *> input_grad_ptr(
+        input_grad.size());
+    for (auto sub_tensor : input_grad) {
+      input_grad_ptr.push_back(&sub_tensor);
+    }
+    int axis = static_cast<int>(this->Attr<int>("axis"));
+    std::vector<paddle::experimental::Tensor *> dx_ptr =
+        this->GetOutputPtr(input_grad_ptr);
+    std::vector<std::string> dx_name = this->GetOutputName(input_grad);
+
+    VLOG(6) << "Runing concat_grad composite func";
+    prim::concat_grad<prim::DescTensor>(input, out_grad, axis, dx_ptr);
+    this->RecoverOutputName(input_grad, dx_name);
+  }
+};
+
```

Comment:
compositeOPMaker should be registered in REGISTER_OPERATOR(concat,

Reply:
done